File: katas/7 kyu/categorize-new-member/solution.js
// categorize-new-member
/* ------------------------------- MY SOLUTION ------------------------------ */
function openOrSenior(data){
  return data.map((x) => x[0] >= 55 && x[1] > 7 ? "Senior" : "Open")
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function openOrSenior(data){
  return data.map(([age, handicap]) => (age > 54 && handicap > 7) ? 'Senior' : 'Open');
}
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Basic tests", () => {
  it("Fixed tests", () => {
    assert.deepEqual(openOrSenior([[45, 12],[55,21],[19, -2],[104, 20]]),['Open', 'Senior', 'Open', 'Senior'])
    assert.deepEqual(openOrSenior([[3, 12],[55,1],[91, -2],[53, 23]]),['Open', 'Open', 'Open', 'Open'])
    assert.deepEqual(openOrSenior([[59, 12],[55,-1],[12, -2],[12, 12]]),['Senior', 'Open', 'Open', 'Open'])
  });
});

File: katas/7 kyu/categorize-new-member/solution.py
# categorize-new-member
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def open_or_senior(data):
  return ["Senior" if x[0] >= 55 and x[1] > 7 else "Open" for x in data ]
# ------------------------------ CLEVER SOLUTION ----------------------------- #
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(open_or_senior([(45, 12),(55,21),(19, -2),(104, 20)]),['Open', 'Senior', 'Open', 'Senior'])
test.assert_equals(open_or_senior([(16, 23),(73,1),(56, 20),(1, -1)]),['Open', 'Open', 'Senior', 'Open'])